src/test/groovy/com/google/protobuf/gradle/ProtobufModelPluginTest.groovy

```diff
@@ -0,0 +1,347 @@
+package com.google.protobuf.gradle
+
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+
+class ProtobufModelPluginTest extends Specification {
+  @Rule
+  final TemporaryFolder testProjectDir = new TemporaryFolder()
+  File buildFile
+
+  def setup() {
+    buildFile = testProjectDir.newFile('build.gradle')
+
+    def pluginClasspathResource = getClass().classLoader.findResource("plugin-classpath.txt")
+    if (pluginClasspathResource == null) {
+      throw new IllegalStateException("Did not find plugin classpath resource, run `testClasses` build task.")
+    }
+
+    def pluginClasspath = pluginClasspathResource.readLines()
+        .collect { it.replace('\\', '\\\\') } // escape backslashes in Windows paths
+        .collect { "'$it'" }
+        .join(", ")
+
+    // Add the logic under test to the test build
+    buildFile << """
+          buildscript {
+            dependencies {
```

Comment:
We could look at making this possible as another PR. This was pull directly from the userguide: https://docs.gradle.org/2.7/userguide/test_kit.html#testKitFunctionalTestSpockClassesTestConfig. I don't see why we couldn't remove the "install" requirement.